tokio-util/src/io/sink_writer.rs

```diff
@@ -0,0 +1,183 @@
+use bytes::Bytes;
+use futures_sink::Sink;
+
+use pin_project_lite::pin_project;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::AsyncWrite;
+
+pin_project! {
+    /// Convert a [`Sink`] of byte chunks into an [`AsyncWrite`].
+    /// Bytes are sent into the sink and the sink is flushed once all bytes are sent. If an error occurs during the sending progress,
+    /// the number of sent but unflushed bytes are saved in case the flushing operation stays unsuccessful.
+    /// For the inverse operation of defining an [`AsyncWrite`] from a [`Sink`] you need to define a [`codec`].
```

Comment:
To make an actual newline in the generated documentation, you have to put an empty line in the doc-comment. So if you intended to have actual newlines, then you are missing some empty lines here. If you did _not_ intend to have actual newlines, then you should wrap each line at 80 characters instead of what you're doing here.